### PR TITLE
Use correct tokenizer for Bedrock/Anthropic LLMs

### PR DESCRIPTION
**Pull request by mrtj**
**Description**

This PR implements the usage of the correct tokenizer in Bedrock LLMs, if using anthropic models.

**Issue:** #11560

**Dependencies:** optional dependency on `anthropic` python library.

**Twitter handle:** jtolgyesi

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
cc @zack-anthropic @3coins @rsgrewal-aws @mani

if we like this approach we should add to ChatBedrock as well

File: libs/langchain/langchain/llms/bedrock.py [hunk 1]
```diff
@@ -393,3 +421,15 @@ def _call(
             return completion
 
         return self._prepare_input_and_invoke(prompt=prompt, stop=stop, **kwargs)
+
+    def get_num_tokens(self, text: str) -> int:
+        if self._model_is_anthropic:
+            return _get_num_tokens_anthropic(text=text)
```
[the remark this answers is not in the document]
Comment (mrtj) [reply libs/langchain/langchain/llms/bedrock.py:427]: Thank you for the observation. Regarding to the anthropic dependency I would like to note that the default implementation of [BaseLanguageModel.get_token_ids](https://github.com/langchain-ai/langchain/blob/c7c03d47095cb7df5a2f90e22cefe710f7a81c61/libs/langchain/langchain/schema/language_model.py#L247-L257) tries to import the `transformers` python library and [raises an uncatched ImportError](https://github.com/langchain-ai/langchain/blob/c7c03d47095cb7df5a2f90e22cefe710f7a81c61/libs/langchain/langchain/schema/language_model.py#L33) if this library is not installed. The token counting in the `transformers` library is not appropriate for Anthropic models, and the `transformers` library is not needed for anything else if you are using Bedrock. So I traded in a useless dependency of `transformers` library to a useful dependency of the `anthropic` library. Also, this ImportError is triggered only if someone is trying to call the `get_token_ids` or `get_num_tokens` methods on the Bedrock LLM, that is not needed for most of the use cases. Notable exceptions are the QA chains that try to count the number of tokens in the documents. When using Bedrock LLM, QA chains do not work without this fix as described in #11560.

**Comment by mrtj** [comment]
> if we like this approach we should add to ChatBedrock as well

If this PR gets a green light, I am happy to integrate the fix also in ChatBedrock. As we are overriding something in BaseLanguageModel, and Bedrock LLM and CharBedrock have different inheritance paths from this base class, I am afraid the implementation has to be duplicated (unless tweaking around with `__mro__` that I would avoid).

File: libs/langchain/langchain/llms/bedrock.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (baskaryan) [reply libs/langchain/langchain/llms/bedrock.py]: you make a great point, forgot that we default to transformers! change makes sense to me, added ChatBedrock support too

**Review by mrtj — COMMENTED** [review]
Looks great to me!

**Comment by baskaryan** [comment]
thanks for the fix @mrtj!

**Comment by stanyq4** [comment]
This fix introduced the following error: `AttributeError: module 'anthropic' has no attribute 'Anthropic'` 

```
File "/opt/homebrew/anaconda3/lib/python3.11/site-packages/langchain/utilities/anthropic.py", line 18, in get_num_tokens_anthropic
    client = _get_anthropic_client()
File "/opt/homebrew/anaconda3/lib/python3.11/site-packages/langchain/utilities/anthropic.py", line 13, in _get_anthropic_client
    return anthropic.Anthropic()
```

Using Langchain v0.0.345

**Comment by stanyq4** [comment]
Update: 

Solved by forcing the latest version of anthropic dependency: 

```
anthropic==0.7.7
```

**Pull request merged**